src/contexts/Photos/Photos.context.tsx

```diff
@@ -131,12 +145,37 @@ export const PhotosContextProvider: React.FC = ({ children }) => {
     }),
   );
 
+  useEffect(() => {
+    checkShouldEnableSync().catch((error) => {
+      reportError(error);
+    });
+  }, []);
+
   useEffect(() => {
     if (syncStatus === PhotosSyncStatus.Completed) {
       setUploadingPhotosItem(null);
     }
   }, [syncStatus]);
 
+  async function checkShouldEnableSync() {
+    const permissionStatus = await permissions.getPermissionsStatus();
+
+    // No permissions, disable the sync
+    if (
+      permissionStatus === MediaLibrary.PermissionStatus.UNDETERMINED ||
+      permissionStatus === MediaLibrary.PermissionStatus.DENIED
+    ) {
+      setSyncEnabled(false);
```

Comment:
I would set this flag on the thenable promise after calling `checkShouldEnableSync` as the method name does not suggest that the responsibility of the function is setting anything rather than just checking if the sync should be enabled or not. 

Therefore, you will be following SRP